libmscore/score.cpp

```diff
@@ -2302,113 +2302,124 @@ void Score::splitStaff(int staffIdx, int splitPoint)
-                              chord->notes().clear();
-                              chord->setTrack(dtrack + voice);
-                              undoAddElement(chord);
+                  if (toDurationElement(e)->tuplet()) {
+                        tupletSrc[voice] = toDurationElement(e)->tuplet();
+                        if (tupletMapping.contains(tupletSrc[voice]))
+                              tupletDst[voice] = tupletMapping[tupletSrc[voice]];
+                        else {
+                              tupletDst[voice] = new Tuplet(*tupletSrc[voice]);
+                              tupletDst[voice]->setTrack(dtrack);
+                              tupletMapping.insert(tupletSrc[voice], tupletDst[voice]);
                               }
-                        Note* nnote = new Note(*note);
-                        nnote->setTrack(dtrack + voice);
-                        chord->add(nnote);
-                        nnote->updateLine();
-                        removeNotes.append(note);
                         }
-                  c->sortNotes();
-                  for (Note* note : removeNotes) {
-                        undoRemoveElement(note);
-                        Chord* chord = note->chord();
-                        if (chord->notes().empty()) {
-                              for (auto sp : spanner()) {
-                                    Slur* slur = toSlur(sp.second);
-                                    if (slur->type() != ElementType::SLUR)
-                                          continue;
-                                    if (slur->startCR() == chord) {
-                                          slur->undoChangeProperty(Pid::TRACK, slur->track()+VOICES);
-                                          for (ScoreElement* ee : slur->linkList()) {
-                                                Slur* lslur = toSlur(ee);
-                                                lslur->setStartElement(0);
-                                                }
+                  else {
+                        tupletSrc[voice] = nullptr;
+                        tupletDst[voice] = nullptr;
+                        }
+
+                  bool createRestDst = true;
+                  bool createRestSrc = false;
+                  TDuration lengthDst = toChordRest(e)->actualDurationType();
+                  TDuration lengthSrc = TDuration();
+
+                  if (e->isChord()) {
+                        Chord* c = toChord(e);
+                        QList<Note*> removeNotes;
+                        for(Note* note : c->notes()) {
+                              if (note->pitch() >= splitPoint)
+                                    continue;
+                              else {
+                                    Chord* chord = toChord(s->element(dtrack + voice));
+                                    Q_ASSERT(!chord || (chord->isChord()));
+                                    if (chord == 0) {
```

Comment:
Use `if (!chord)`.

Reply:
You're right. But as a matter of fact, this was just a copy of the existing code :-). Will be changed nevertheless.

Reply:
Probably `!chord || (chord->isChord()`

Reply:
<code>chord->isChord()</code> is not needed here since <code>chord</code> comes from an <code>element</code> from which we know it is a <code>ChordRest</code> already. What we don't k now is whether it has a <code>Chord </code> on that track.

Reply:
<code>(chord->isChord())</code> is not needed since we know it is taken from an element from which we know it comes from a <code>ChordRest</code> segment already (we are only looking for this kind of segments). The element could be a <code>Rest</code> of course but we are already inside an if branch were it was checked whether the element was a <code>Chord</code> or not.
What we don't know whether there is a <code>Chord</code> for the requested track and that's what is tested here.